pkg/aws/ec2/api/wrapper.go

```diff
@@ -639,6 +657,38 @@ func (e *ec2Wrapper) DescribeNetworkInterfaces(input *ec2.DescribeNetworkInterfa
 	return describeNetworkInterfacesOutput, err
 }
 
+// DescribeNetworkInterfacesPages returns network interfaces that match the filters specified in the input with MaxResult set to 1000(max value)
+// This API is used during periodic ENI cleanup routine and trunk initialization to list all network interfaces that match the given filters (vpc-id or subnet-id, and tag)
+// Only required fields, network interface ID and tag set, is populated to avoid consuming extra memory
+func (e *ec2Wrapper) DescribeNetworkInterfacesPages(input *ec2.DescribeNetworkInterfacesInput) ([]*ec2.NetworkInterface, error) {
+	var networkInterfaces []*ec2.NetworkInterface
+	input.MaxResults = aws.Int64(config.DescribeNetworkInterfacesMaxResults)
+
+	start := time.Now()
+	if err := e.userServiceClient.DescribeNetworkInterfacesPages(input, func(output *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+		ec2APICallCnt.Inc()
+		ec2DescribeNetworkInterfacesPagesAPICallCnt.Inc()
+		//Currently only network interface ID and the tag set is require, only add required details to avoid consuming extra memory
```

Comment:
What does `avoid consuming extra memory` mean?

Reply:
This can explained in the comment too. It will help us in future.

Reply:
Ack, I can add more info here in a follow-up PR.